### ISSUE TITLE
gulpfile.babel.js not used if parent directory has a gulpfile.js

### ISSUE THREAD

**Issue by nrser** [gulpjs/gulp#1823]
**What were you expecting to happen?**

for directory structure like

```
project-dir/
    submodules/
       my-module/
           gulpfile.babel.js
    gulpfile.js
```

when i run `gulp` in the `project-dir/submodules/my-module` directory i expected it to use `project-dir/submodules/my-module/gulpfile.babel.js`.

**What actually happened?**

it uses `project_dir/gulpfile.js`.

if i change the name of `project-dir/gulpfile.js` to `project-dir/gulpfile.babel.js` it works as expected.

(as the example directory structure suggests, this problem cropped up when changing a submodule in a project to use `gulpfile.babel.js` while the main repo was still using `gulpfile.js`)

**Please post a sample of your gulpfile (preferably reduced to just the bit that's not working)**

it doesn't matter what's in the `gulpfile`. you can reproduce via: 

```
mkdir project-dir
cd project-dir
npm init -y
npm install --save-dev gulp
touch gulpfile.js
mkdir -p submodules/my-module
cd submodules/my-module
npm init -y
npm install --save-dev gulp babel-register
touch gulpfile.babel.js
gulp -T
```

output shows gulp walking up the directory tree to `gulpfile.js`:

```
[21:49:03] Working directory changed to /private/tmp/project-dir
[21:49:03] Using gulpfile /private/tmp/project-dir/gulpfile.js
[21:49:04] Tasks for /private/tmp/project-dir/gulpfile.js
```

then if you

```
mv ../../gulpfile.js ../../gulpfile.babel.js
gulp -T
```

you'll see it use the correct file:

```
[21:50:08] Requiring external module babel-register
[21:50:08] Using gulpfile /private/tmp/project-dir/submodules/my-module/gulpfile.babel.js
[21:50:08] Tasks for /private/tmp/project-dir/submodules/my-module/gulpfile.babel.js
```

**What version of gulp are you using?**

```
$ gulp --version
[21:36:13] Requiring external module babel-register
[21:36:13] CLI version 3.9.1
[21:36:13] Local version 3.9.1
```

**What versions of npm and node are you using?**

```
$ npm --version
3.10.3

$ node --version
v6.6.0
```

**Comment by MGBR** [comment]
I'm having the same issue. 

Also, if the parent dir doesn't have a gulpfile, gulp returns an error `Local gulp not found in <parent-dir>`. In my project, it's looking for the gulpfile three levels up the file tree from where I'm running gulp

**Comment by contra** [comment]
@tkellen looks like a liftoff bug

**Comment by phated** [comment]
It's actually a bug in the updated version of `findup-sync` - I'll open an issue over there

**Comment by phated** [comment]
Issue js-cli/node-findup-sync#38 and failing test js-cli/node-findup-sync#39

**Comment by jonschlinkert** [comment]
> It's actually a bug in the updated version of findup-sync

since findup-sync is supposed to find "the first file matching a given pattern in the current directory or the nearest ancestor directory", I'm not sure if this is related to findup-sync. IMHO it sounds like liftoff is failing to find `gulpfile.babel.js` in this case specifically because findup-sync is doing what it's supposed to do. 

Wasn't the big pr in liftoff related to this? I thought it was for resolving multiple config files

**Comment by jonschlinkert** [comment]
@doowb's [suggestion](https://github.com/js-cli/node-findup-sync/pull/39#issuecomment-251536726) to use a brace pattern is a really elegant one that seems like it would work for the time being. 

We could also implement a "breadth-first" search in findup-sync, so that when multiple globs are used each pattern is checked in the given directory before moving onto the next directory. Currently a "depth-first" search is done.

**Comment by phated** [comment]
@jonschlinkert I'm not opposed to that solution but I'm pretty sure this was another bug introduced by the findup-sync refactor. The PR you are referring to was a new feature, unrelated to this.

**Comment by dWhisper** [comment]
Having the same issue that @MGBR posted, and was thinking I was actually going insane because I couldn't figure out what changed between this morning and this afternoon when it started to happen (after I updated some other stuff).

No directories that are set up with a gulpfile.babel.js are working since the directories above (my home and a Code directory I use to organize) have any gulp / npm dependencies. 

It seems to be skipping out on stuff like .babelrc presets entirely; for my projects, even doing a gulp -v is skipping the babel-register step.

**Comment by phated** [comment]
This was just fixed in `findup-sync@0.4.3` - please upgrade your dependency tree.
